tests/test_dragon_installer.py

```diff
@@ -0,0 +1,519 @@
+        pytest.param("dragon-0.9", "py3.8", False, True, id="0.9,python 3.8,CrayEX"),
+        pytest.param("dragon-0.9", "py3.9", True, True, id="0.9,python 3.9,CrayEX"),
+        pytest.param("dragon-0.9", "py3.10", True, True, id="0.9,python 3.10,CrayEX"),
+        pytest.param("dragon-0.9", "py3.11", True, True, id="0.9,python 3.11,CrayEX"),
+        pytest.param("dragon-0.9", "py3.12", False, True, id="0.9,python 3.12,CrayEX"),
+        # add a couple variants for a dragon version that isn't in the asset list
+        pytest.param("dragon-0.7", "py3.9", False, False, id="0.7,python 3.9"),
+        pytest.param("dragon-0.7", "py3.9", False, True, id="0.7,python 3.9,CrayEX"),
+    ],
+)
+def test_retrieve_asset_info(
+    asset_list: t.Dict[str, GitReleaseAsset],
+    monkeypatch: pytest.MonkeyPatch,
+    dragon_pin: str,
+    pyv: str,
+    is_found: bool,
+    is_crayex: bool,
+) -> None:
+    """Verify that an asset list is filtered correctly based on the python
+    version, platform (e.g. CrayEX, !CrayEx), and target dragon pin"""
+
+    with monkeypatch.context() as ctx:
+        ctx.setattr(
+            smartsim._core.entrypoints.dragon_install,
+            "python_version",
+            lambda: pyv,
+        )
+        ctx.setattr(
+            smartsim._core.entrypoints.dragon_install,
+            "is_crayex_platform",
+            lambda: is_crayex,
+        )
+        ctx.setattr(
+            smartsim._core.entrypoints.dragon_install,
+            "dragon_pin",
+            lambda: dragon_pin,
+        )
+        # avoid hitting github API
+        ctx.setattr(
+            smartsim._core.entrypoints.dragon_install,
+            "_get_release_assets",
+            lambda: asset_list,
+        )
+
+        if is_found:
+            chosen_asset = retrieve_asset_info()
+
+            assert chosen_asset
+            assert pyv in chosen_asset.name
+            assert dragon_pin in chosen_asset.name
+
+            if is_crayex:
+                assert "crayex" in chosen_asset.name.lower()
+        else:
+            with pytest.raises(SmartSimCLIActionCancelled):
+                retrieve_asset_info()
+
+
+def test_check_for_utility_missing(test_dir: str) -> None:
+    """Ensure that looking for a missing utility doesn't raise an exception"""
+    ld_config = pathlib.Path(test_dir) / "ldconfig"
+
+    utility = check_for_utility(ld_config)
+
+    assert not utility
+
+
+def test_check_for_utility_exists(test_dir: str) -> None:
+    """Ensure that looking for an existing utility returns a non-empty path"""
+    utility = check_for_utility("ls")
+
+    assert utility
+
+
+def test_is_crayex_missing_ldconfig(monkeypatch: pytest.MonkeyPatch) -> None:
+    """Ensure the cray ex platform check doesn't fail when ldconfig isn't
+    available for use"""
+
+    def mock_util_check(util: str) -> bool:
+        if util == "ldconfig":
+            return False
+        return True
```

Comment:
Not wrong, just be warned: There is an implict change of type for `smartsim._core.entrypoints.dragon_install.check_for_utility` from `t.Callable[[str], str]` to `t.Callable[[str], bool]` with this patch fn.

Reply:
Great eye. Fixed!